### PR TITLE
[FIX] mail_bot: odoobot waits 1 week between notification requests

### PR DESCRIPTION
**Pull request by jhk-odoo**
Steps to reproduce:
- install mail_bot
- wait for odoobot to ask you if you want to receive notifications
- do not select "grant" or "deny", click the closing cross

Previous behavior:
Odoobot would consider that you denied the request

Current behavior:
Odoobot waits 1 week and will ask you again

opw-2186299

### CODE REVIEW

**Comment by jhk-odoo** [comment]
@jpp-odoo

File: addons/mail_bot/static/src/js/mailbot_service.js [hunk 1]
```diff
@@ -13,7 +14,9 @@ var MailBotService =  AbstractService.extend({
      */
     start: function () {
         this._hasRequest = (window.Notification && window.Notification.permission === "default") || false;
-        if ('odoobot_initialized' in session && ! session.odoobot_initialized) {
+        var lastRequestTime = localStorage.getItem('odoobot_notification_last_request_time');
```
Comment (kebeclibre) [inline addons/mail_bot/static/src/js/mailbot_service.js]: Make sure the timestamp you use is universal, that is in UTC i guess.
Use some kind of variation of session.getTZOffset

File: addons/mail_bot/static/src/js/mailbot_service.js [hunk 2]
```diff
@@ -60,6 +63,7 @@ var MailBotService =  AbstractService.extend({
      * Called when user either accepts or refuses push notifications.
      */
     removeRequest: function () {
+        localStorage.setItem('odoobot_notification_last_request_time', new Date().getTime());
```
Comment (kebeclibre) [inline addons/mail_bot/static/src/js/mailbot_service.js]: same here

File: addons/mail_bot/static/src/js/mailbot_service.js (hunk 1 above)
Comment (nim-odoo) [reply addons/mail_bot/static/src/js/mailbot_service.js]: Not sure this is useful. We are interested in a difference between 2 datetimes; whether they are in UTC or TZ-aware is not relevant as long as they are consistent. In this case, `new Date().getTime()` always returns the UTC milliseconds, so it looks good.

File: addons/mail_bot/static/src/js/mailbot_service.js (hunk 2 above)
Comment (nim-odoo) [reply addons/mail_bot/static/src/js/mailbot_service.js]: Maybe add some DB identifier in your local storage key? If you are connected to several DB, the keys may collide.

**Comment by jhk-odoo** [comment]
I prefixed with the db_name
for the test: i'm not sure how to mock the localstorage

**Comment by jhk-odoo** [comment]
bump

**Comment by jhk-odoo** [comment]
updated to fit previous discussion

File: addons/mail_bot/static/src/js/mailbot_service.js [hunk 3]
```diff
@@ -13,9 +14,13 @@ var MailBotService =  AbstractService.extend({
      */
     start: function () {
         this._hasRequest = (window.Notification && window.Notification.permission === "default") || false;
+        var lastRequestTime = localStorage.getItem(session.db + '_odoobot_notification_last_request_time');
```
Comment (nim-odoo) [inline addons/mail_bot/static/src/js/mailbot_service.js]: we can get rid of `session.db`, not necessary

**Pull request closed without merging**